addons/account_edi/models/account_edi_document.py

```diff
@@ -59,19 +72,60 @@ def _prepare_jobs(self):
                 continue
 
             key = (edi_format, edi_doc.state, doc_type, move.company_id)
+            if not to_process.get(key, None):
+                to_process[key] = self.env['account.edi.document']
+            to_process[key] |= edi_doc
+
+        # Order payments/invoice and create batches.
+        result = []
+        payments = []
+        for key, documents in to_process.items():
+            edi_format, state, doc_type, company_id = key
+            target = result if doc_type == 'invoice' else payments
             if edi_format._support_batching():
-                if not batches.get(key, None):
-                    batches[key] = self.env['account.edi.document']
-                batches[key] |= edi_doc
+                target.append((documents, doc_type))
             else:
-                to_process.append((key, edi_doc))
-        to_process.extend(batches.items())
-        return to_process
+                target.extend((doc, doc_type) for doc in documents)
+        result.extend(payments)
+        return result
+
+    @api.model
+    def _convert_to_old_jobs_format(self, jobs):
+        """ See '_prepare_jobs' :
+        Old format : ((edi_format, state, doc_type, company_id), documents)
+        Since edi_format, state and company_id can be deduced from documents, this is redundant and more prone to unexpected behaviours.
+        New format : (doc_type, documents).
+
+        However, for backward compatibility of 'process_jobs', we need a way to convert back to the old format.
+        """
```

Comment:
Related to comment https://github.com/odoo/odoo/pull/62305/files#r531418887 (just a few lines bellow)

I would still like to keep the new format for the jobs, the reason is explained in this docstring.
Maybe in master we can change the signature of '_process_jobs' and remove this method.
@smetl would like your opinion on this.